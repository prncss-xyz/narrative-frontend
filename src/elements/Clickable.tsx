import { Box } from "./basics";

export function Clickable({
  onClick,
  children,
  disabled,
  ...props
}: {
  onClick: () => void;
  children: React.ReactNode;
  disabled?: boolean;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box>
      <Box
        as="button"
        backgroundColor="inherit"
        borderStyle="none"
        padding="0px"
        height="100%"
        cursor={disabled ? undefined : "pointer"}
        disabled={disabled}
        onClick={onClick}
        {...props}
      >
        {children}
      </Box>
    </Box>
  );
}
