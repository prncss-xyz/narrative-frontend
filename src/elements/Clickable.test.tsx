import { expect, it } from "vitest";
import renderer from "react-test-renderer";
import { Clickable } from "./Clickable";

it("renders correctly", () => {
  const tree = renderer.create(<Clickable onClick={console.log}>Test</Clickable>).toJSON();
  expect(tree).toMatchSnapshot();
});
