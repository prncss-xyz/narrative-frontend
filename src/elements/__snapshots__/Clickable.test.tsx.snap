// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<button
  className="css-17pgxs5-Clickable"
  onClick={[Function]}
>
  Test
</button>
`;
